lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -1,5 +1,106 @@
 define((require, exports, module) ->
   LatexParsingContext = require("ace/ext/papeeria/latex_parsing_context")
+  TokenIterator = require("ace/token_iterator").TokenIterator
+  Range = require("ace/range").Range
+
+  getEquationRangeHandler = (editor) ->
+    erh = {
+      BEGIN_EQUATION_TOKEN_SEQUENCE: [
+        {
+          type: "storage.type"
+          value: "\\begin"
+        }
+        {
+          type: "lparen"
+          value: "{"
+        }
+        {
+          type: "variable.parameter"
+          value: "equation"
+        }
+        {
+          type: "rparen"
+          value: "}"
+        }
+      ]
+      END_EQUATION_TOKEN_SEQUENCE: [
+        {
+          type: "storage.type"
+          value: "\\end"
+        }
+        {
+          type: "lparen"
+          value: "{"
+        }
+        {
+          type: "variable.parameter"
+          value: "equation"
+        }
+        {
+          type: "rparen"
+          value: "}"
+        }
+      ]
+
+      compareTokens: (token1, token2) ->
```

Comment:
pls rename it to `equalTokens`, because compare normally means establishing order (and returns -1, 0, -1)